apps/table-frontend/src/app/components/from-research.tsx

```diff
@@ -0,0 +1,56 @@
+import React, { useEffect } from 'react';
+import { getFile } from './get-file-info';
+import './from-research.css';
+import {
+  ParsedArticleParagraphStandalone,
+  ArticleParagraphBacksUpClaim,
+} from '@foodmedicine/interfaces';
+
+interface Paragraph {
+  url: string;
+  body: string;
+}
+
+export interface FromResearchProps {
+  fileName: string;
+}
+
+function LinkedShortParagraph(props: { paragraph: Paragraph }) {
+  function shortenText(text: string) {
+    return text.length > 400 ? text.substring(0, 400) + '...' : text;
```

Comment:
why not just set a variable shortenedText = <this logic>, and in the html returned, use that shortenedText directly instead of having to call shortenText function

Reply:
Changed to that